go/controller/cluster.go

```diff
@@ -95,6 +95,25 @@ func getPodsTotalRequestsAndLimits(podList *v1.PodList) (reqs v1.ResourceList, l
 	return
 }
 
+func syncNodesIdleResource(podList *v1.PodList, nodesCPUIdleMilli map[string]int64, nodesMemoryFreeMega map[string]int64) (err error) {
+	for _, pod := range podList.Items {
+		nodeName := pod.Spec.NodeName
+		if nodeName == "" {
+			continue
+		}
+		for _, container := range pod.Spec.Containers {
```

Comment:
Should we check only pending and running pods?

Reply:
Only Terminating and Running Pod will take up the resource.

Reply:
Oh ok, so maybe we need to check only "Terminating and Running" pod? :)

Reply:
I think so, follow the comment https://github.com/kubernetes/kubernetes/blob/v1.6.2/pkg/api/v1/types.go#L2331, if `NodeName` is non-empty, the Pod will be assigned to the Node, mean take up the resuore on the Node.

Reply:
I see, so it's not necessary to check pod status, since as long as `pod.Spec.NodeName` is not empty, it's assigned to a node, and occupying resource.